floats.go

```diff
@@ -69,9 +69,8 @@ func AddScaledTo(dst, y []float64, alpha float64, s []float64) []float64 {
 	if len(dst) != len(s) || len(dst) != len(y) {
 		panic("floats: lengths of slices do not match")
 	}
-	for i, val := range s {
-		dst[i] = y[i] + alpha*val
-	}
+	copy(dst, y)
```

Comment:
If I'm understanding you correctly, there is no reason to have two versions of DaxpyUnitary. The blas/native code will call it as `DaxpyUnitary(alpha, x, y, y)` and the code here will call it as `DaxpyUnitary(alpha, x, y, z)`.